torch/_dynamo/backends/distributed.py

```diff
@@ -311,8 +313,43 @@ def run_node(self, n: Node) -> Any:
             # be FakeTensors already since Dynamo would have made them FakeTensors in the
             # non-DDP flow.  However, the parameters are _not_ expected to be FakeTensors,
             # since this wrapping happens during compilation
+
+            # Note: Returning Fake Tensors on First AOT Autograd Call
+            #
+            # Inductor will optimize strides of outputs when it deems it profitable.
+            # For instance, converting to channels last. When we split the graph here
+            # into multiple inductor compilations, we need to make sure that the
+            # output strides of one compilation is appropriately passed to the subsequent
+            # compilations. However, the mapping from inductor output to dynamo output
+            # is non-trivial due to aot_autograd's deduping, de-aliasing, mutation, re-writing,
+            # subclass handling, etc. In order to replay all this logic we set a flag such that
+            # the first invocation of inductor in aot_autograd will return Fake Tensors with
+            # appropriate strides. Then, all of aot autograd's runtime logic is replayed.
+            # This gives us the appropriately strided outputs here which will reflect runtime strides.
+
+            class FakifyGuard:
+                def __init__(self):
+                    self.tc = torch._guards.TracingContext.try_get()
+                    assert self.tc
+                    torch._guards.TracingContext.try_get().fakify_first_call = True
+
+                def __del__(self):
+                    self.tc.fakify_first_call = False
+
+            # For aot_eager and other backends, tracing context is not set
+            has_tracing_context = torch._guards.TracingContext.try_get() is not None
+            if has_tracing_context:
+                g = FakifyGuard()
+
+            from torch._dynamo.utils import counters
+
+            init = counters["aot_autograd"]["total"]
             compiled_submod_real = self.compile_submod(real_mod, new_args, kwargs)
 
+            # TODO - better way of doing this?
+            # Only aot autograd handles fakifying first call
```

Comment:
yeah... good call.

Alternatively - do we have a way to detect if we are compiling for inductor? If we do, that might be easier than trying to add support for this for any backend that happens to use aot autograd (hopefully most other backends don't change output striding)

Reply:
Oh maybe we could even branch here on `report_output_strides()`, since that's the only case that matters anyway

Reply:
At this point we have already passed in fakify_first_outputs to aot_autograd. If it were a non inductor backend then we'd still have to invoke it now so that the first output wouldn't be fake.